jubatus/core/classifier/nearest_neighbor_classifier.cpp

```diff
@@ -165,7 +177,7 @@ bool nearest_neighbor_classifier::delete_label(const std::string& label) {
 }
 
 void nearest_neighbor_classifier::clear() {
-  nearest_neighbor_engine_->clear();
+  nearest_neighbor_engine_->clear();  // lock acquired inside
   labels_.clear();
   if (unlearner_) {
     unlearner_->clear();
```

Comment:
Unlike ``labels_``, unlearner is not a thread safe component. We should take lock for it I think.
(maybe another issue)